### PR TITLE
[asl] release acquire

### PR DESCRIPTION
**Pull request by maranget**
Implement release and acquire properties on loads (`LDAR` and `LDAPR`), stores (`STLR`), swaps (`SWPA`, `SWPL`, `SWPAL`) and compare and swaps (`CASA`, `CASL`, `CASAL`).

### CODE REVIEW

File: herd/AArch64Sem.ml [hunk 1]
```diff
@@ -563,7 +566,7 @@ module Make
           (fun _ -> mk_fault (Some a) dir an ii ft (Some "EL0"))  ii >>! B.Exit
 
       let an_xpte =
```
Comment (HadrienRenaud) [inline herd/AArch64Sem.ml:568]: Maybe `an_xpte` and `an_pte` could go into `AArch64Annot.ml`?

**Review by HadrienRenaud — APPROVED** [review]
LGTM

File: herd/AArch64Sem.ml (hunk 1 above)
Comment (maranget) [reply herd/AArch64Sem.ml:568]: Would those function also be called AArch64ASLSem.ml?

Comment (maranget) [reply herd/AArch64Sem.ml:568]: Thinking twice, I'd rather keep those functions here, close to their calling sites.

Comment (HadrienRenaud) [reply herd/AArch64Sem.ml:568]: All right

**Pull request merged**